src/github_repo_request_local.py

```diff
@@ -346,26 +345,33 @@ def detect_test_runners2(repo_path):
     else:
         csv_file_path = repo_root / input_file
 
-        if csv_file_path.exists():
-            df = pd.read_csv(csv_file_path)
+        if not csv_file_path.exists():
             logger.info(
-                "Initialising the columns : 'clone_status', 'testfilecountlocal', 'last_commit_hash'"
+                f"The input file has not been found at {csv_file_path}."
+                f" Running the script `initial_data_preparation.py` to "
+                f"create the required dataframe."
             )
-            df["clone_status"] = None  # Initialise the clone status column
-            df["testfilecountlocal"] = -1  # Initialise if first run
-            df["last_commit_hash"] = None
-            logger.info(f"Creating and initialising columns : {test_runners.keys()}")
-            for runner in test_runners.keys():
-                df[f"{runner}_dependency_patterns"] = 0
-                df[f"{runner}_config_files"] = 0
-                df[f"{runner}_file_patterns"] = 0
-
-        else:
-            logger.error(
-                f"The input file has not been found at {csv_file_path}. Exiting..."
+            subprocess.run(
+                ["python", repo_root / "utils" / "initial_data_preparation.py"]
```

Comment:
This is a reasonable approach (to call the other script if the data doesn't exist); however it complicates the logic slightly more than necessary e.g. what happens if it returns an error? Another approach that should work at least as well is to log the mismatch in expectation (this script expects the csv file to exist) as with a log level of error and then do a `sys,exit(...)`  probably with a return value of 1. 

I'd suggest reverting most of this recent change, however please incorporate the additional information you added during this change along the lines of the following message:
```
f"The input file has not been found at {csv_file_path}."
                f" Please run the script `initial_data_preparation.py` to "
                f"create the required dataframe. Exiting"
```